haystack/preview/components/rankers/meta_field.py

```diff
@@ -0,0 +1,186 @@
+        3. Returning the top-k documents.
+
+        :param query: Not used in practice (so can be left blank), as this ranker does not perform sorting based on semantic closeness of documents to the query.
+        :param documents: Documents provided for ranking.
+        :param top_k: (optional) How many documents to return at the end. If not provided, all documents will be returned.
+        """
+        if not documents:
+            return {"documents": []}
+
+        if top_k is None:
+            top_k = self.top_k
+        elif top_k <= 0:
+            raise ValueError(f"top_k must be > 0, but got {top_k}")
+
+        try:
+            sorted_by_metadata = sorted(documents, key=lambda doc: doc.meta[self.metadata_field], reverse=True)
+        except KeyError:
+            raise ComponentError(
+                """
+                Param <metadata_field> was set to '{}' but document(s) {} do not contain this metadata key.\n
+                Please double-check the names of existing metadata fields of your documents \n
+                and set <metadata_field> to the name of the field that contains the metadata you want to rank by.
+                """.format(
+                    self.metadata_field, ",".join([doc.id for doc in documents if self.metadata_field not in doc.meta])
+                )
+            )
+
+        if self.weight > 0:
+            sorted_documents = self._merge_scores(documents, sorted_by_metadata)
+            return {"documents": sorted_documents[:top_k]}
+        else:
+            return {"documents": sorted_by_metadata[:top_k]}
+
+    def _merge_scores(self, documents: List[Document], sorted_documents: List[Document]) -> List[Document]:
+        """
+        Merge scores for documents sorted both by content and by metadata field.
+        """
+        scores_map: Dict = defaultdict(int)
+
+        # scores coming from the retriever/ranker are weighted with (1 - self.weight)
+        for i, doc in enumerate(documents):
+            if self.ranking_mode == "reciprocal_rank_fusion":
+                scores_map[doc.id] += self._calculate_rrf(rank=i) * (1 - self.weight)
+            elif self.ranking_mode == "linear_score":
+                score = float(0)
+                if doc.score is None:
+                    warnings.warn("The score was not provided; defaulting to 0")
+                elif doc.score < 0 or doc.score > 1:
+                    warnings.warn(
+                        "The score {} for document {} is outside the [0,1] range; defaulting to 0".format(
+                            doc.score, doc.id
+                        )
+                    )
+                else:
+                    score = doc.score
+
+                scores_map[doc.id] += score * (1 - self.weight)
+
+        # scores coming from the metadata field are weighted with self.weight
+        for i, doc in enumerate(sorted_documents):
```

Comment:
Let's perhaps have separate loops for each ranking mode. And also raise and exception for unknown ranking mode before we even start merging

Reply:
That makes sense, I added the suggestion. The exception for unknown ranking mode is present at line 70, in the `__init__` method, should I move that to the `run` function?